### PR TITLE
Change: Split building build and production container images

### PR DESCRIPTION
**Pull request by bjoernricks**
**What**:

Split building build and production container images

**Why**:

The build image really rarely needs a rebuild. Therefore split this job
into an own workflow and only rebuild if the corresponding dockerfile or
workflow has changed.

**How**:

**Checklist**:

- [ ] Tests
- [ ] PR merge commit message adjusted

### CODE REVIEW

**Comment by Dexus** [comment]
I would recommend to rebuild the build image at least once all 2 weeks via schedule, because new packages will else not used for the builds.

**Comment by bjoernricks** [comment]
> I would recommend to rebuild the build image at least once all 2 weeks via schedule, because new packages will else not used for the builds.

I've also thought about that. Could you check if the schedule fits now? not sure if i got that cron syntax right

File: .github/workflows/build-container.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      - .docker/build.Dockerfile
+  pull_request:
+    branches: [ main, stable, oldstable ]
+    paths:
+      - .github/workflows/build-container.yml
+      - .docker/build.Dockerfile
+  workflow_dispatch:
+  schedule:
+    # rebuild image every sunday
+    - cron: 0 0 * * 0
```
Comment (Dexus) [inline .github/workflows/build-container.yml:18]:
```suggestion
    - cron: '0 0 * * 0'
```

**Review by Dexus — CHANGES_REQUESTED** [review]

**Review by Dexus — APPROVED** [review]

**Comment by Dexus** [comment]
LGTM

**Pull request merged**